### DISCUSSION TITLE
Support and Example for publishing M1 and Intel nativeImage

### DISCUSSION THREAD

**Discussion by yschimke** [jreleaser/jreleaser#801] [Ideas]
From https://twitter.com/yschimke/status/1506246058531336199

Ideally confirmation of support, and an example github CI build producing both and publishing to homebrew and others.

**Comment by aalmiray** [comment]
Well, at the moment GitHub does not support building on Apple Silicon hosted runners. You would have to configure a self-hosted runner to make it happen on CI. The produced artifact can be configured just like any other artifact but make sure to use the `osx-aarch_64` platform. You may use [platform replacements](https://jreleaser.org/guide/latest/configuration/platform.html) to map `osx-aarch_64` to `osx-aarch64` or even `darwin-aarch64`.

An alternative to self-hosted runners is to build the Apple Silicon artifact locally, collect all other artifacts generated on GitHub, and post a release from your local M1 environment.

**Reply by yschimke** [reply]
Thanks, makes sense.

**Reply by yschimke** [reply]
A request for Github to support those runners is probably the first step.
